src/main/java/org/spongepowered/common/util/OptionalUtils.java

```diff
@@ -24,17 +24,15 @@
  */
 package org.spongepowered.common.util;
 
-import java.util.Collection;
-import java.util.Collections;
 import java.util.Optional;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
 
 public class OptionalUtils {
 
-    public static <E> Collection<E> asSet(Optional<E> opt) {
-        if (opt.isPresent()) {
-            return Collections.singleton(opt.get());
-        }
-        return Collections.emptySet();
+    public static <E> Set<E> asSet(Optional<E> opt) {
```

Comment:
Why are you changing this to `ImmutableSet`?

Reply:
It was in the original PR - 
https://github.com/SpongePowered/SpongeCommon/pull/793/files#diff-c2ae1593cc5f5d014b6d276969856f4cR35

That said, it does make sense that a method labelled “asSet” returns a set rather than a collection.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#singleton-T-

Reply:
I was referring to the signature change.

Reply:
I seem to have clicked on the wrong line when commenting - the signature change is fine, I'm referring to the change from `Collections` to `ImmutableSet`.

Reply:
I know you were, considering the signature didn’t change to an immutable set. My second comment was about the signature, not about the contents, I just wasn’t clear.

Reply:
:tada: